server/example/websocket-event-manager.js

```diff
@@ -1,23 +1,24 @@
-const logEl = document.getElementById("log");
-const log = (msg) => logEl.value += `${msg}\n`;
-
+/**
+ * Client-side WebSocket event manager.
+ * The `onLog` property of an instance can be set as a function receiving a log string.
+ */
 class WebSocketEventManager {
   constructor(url, onOpen) {
     this.events = {};
     this.ws = new WebSocket(url);
 
     this.ws.onopen = () => {
-      log(`WebSocket opened at ${url}`);
+      if(this.onLog) this.onLog(`WebSocket opened at ${url}`);
       if(onOpen) onOpen();
     };
 
     this.ws.onmessage = (message) => {
       message = JSON.parse(message.data);
       if(!message.event) {
-        log(`Invalid message received: ${message}`);
+        if(this.onLog) this.onLog(`Invalid message received: ${message}`);
         return;
       }
-      log(`Event ${message.event} received: ${JSON.stringify(message.data)}`);
+      if(this.onLog) this.onLog(`Event ${message.event} received: ${JSON.stringify(message.data)}`);
```

Comment:
This line is a little difficult to read as a single line, but if it works for you go for it.